subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaPlugin.java

```diff
@@ -70,7 +71,14 @@ private static void configureScaladoc(final Project project, final SourceSet mai
                 files.from(main.getCompileClasspath());
                 return files;
             });
-            scalaDoc.setSource(main.getExtensions().getByType(ScalaSourceDirectorySet.class));
+
+            boolean isScala3 = ScalaRuntimeHelper.findScalaJar(main.getCompileClasspath().getFiles(), "library_3") != null;
```

Comment:
This is something that I missed earlier and which is an actual blocker for releasing the support.
It actually causes test failures because in some cases, we end up eagerly resolving the `compileClasspath`, so early that it prevents adding any dependencies ...

This will require more work and so effectively postpone Scaladoc support for Scala 3 to Gradle 7.4

Reply:
To be clear, fixing this is actually non trivial and may not even be possible with the existing lazy APIs of Gradle. So I will be following up and updating here as soon as I dig deeper.

Reply:
Also, the problem is that we want to have the autoconfiguring scala plugin for scala2 and 3, which differs in overall API and makes the whole process harder. I was thinking that maybe we should create separate plugins, like `scala` and `scala3` so there won't be all these classpath guessings.